dslink-scala-nest/src/main/scala/com/uralian/nest/service/NestHttpClient.scala

```diff
@@ -74,4 +78,29 @@ class NestHttpClient(config: NestClientConfig) {
   def httpGet[T: Manifest](query: String)
                           (implicit at: AccessToken, as: ResponseAs[T, Nothing], ec: ExecutionContext): Future[T] =
     httpGet(uri"${config.apiUrl}/$query")
+
+  val log = LoggerFactory.getLogger(getClass)
+
+  /**
+    *
+    * @param at
+    * @param ec
+    * @return
+    */
+  def httpStream()(implicit at: AccessToken, ec: ExecutionContext): Future[Source[ByteString, NotUsed]] = {
+
+    val request = sttp.auth.bearer(at.token).header("Accept", "text/event-stream")
+    log.info(s"sending ${at.token}")
+    val response = request.get(uri"${config.apiUrl}/devices/thermostats/").response(asStream[Source[ByteString, NotUsed]]).send()
+    response map {
+      case Response(Right(body), _, _, _, _) => {
+
+        body
```

Comment:
Other methods in `NestHttpClient` allow type parameter T, so they can automatically convert the response to T. Should we also allow this here for consistency, instead of returning `Source[ByteString, _]`? Also, should we hardcode `/devices/thermostats`? I believe we can also stream other fragments of the full response? Anyway, this is minor, but here's my stab at this method, we can discuss it later:

```scala
 def httpGetStream[T: Manifest](uri: Uri)
                                (implicit at: AccessToken, parser: String => T, ec: ExecutionContext) = {
    val request = sttp.auth.bearer(at.token).contentType(MediaTypes.EventStream)
    val response = request.get(uri).response(asStream[Source[ByteString, Any]]).send()
    val source = response map {
      case Response(Right(body), _, _, _, _) => body
      case Response(Left(bytes), _, _, _, _) => throw new RuntimeException(new String(bytes))
    } map { src =>
      src.map(s => parser(s.utf8String))
    }

    source
  }
```

Reply:
Let me take a look into this...